I played around with this a bit locally and noticed the typings require all three arguments to be passed:

```typescript
// test.ts
import getOpts from "get-options";
console.log(getOpts(["--help", "--verbose"], {
    "--help, -?": "",
    "--verbose":  "",
}));
```

```console
λ tsc test.ts
test.ts:2:13 - error TS2554: Expected 3 arguments, but got 2.

2 console.log(getOpts(["--help", "--verbose"], {
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
3     "--help, -?": "",
  ~~~~~~~~~~~~~~~~~~~~~
4     "--verbose":  "",
  ~~~~~~~~~~~~~~~~~~~~~
5 }));
  ~~

  node_modules/get-options/index.d.ts:20:2
    20  config: Config
        ~~~~~~~~~~~~~~
    An argument for 'config' was not provided.
```

Both the second and third arguments are actually optional (`optdef` should probably have a default value in the function header to communicate this better...). Basically, if you don't define anything specific, it [tries to do the right thing](https://github.com/Alhadis/GetOptions/blob/master/docs/anonymous-options.md) with its input.

Would it be possible to amend the typings to reflect this?

--- a/index.d.ts
+++ b/index.d.ts
@@ -16,8 +16,8 @@ export declare type Config = {
 };
 export declare function getOpts(
 	input: any[],
-	optdef: string | { [key: string]: string },
-	config: Config
+	optdef?: string | { [key: string]: string },
+	config?: Config
 ): Options;
 
 export default getOpts;

--- a/index.d.ts
+++ b/index.d.ts
@@ -1,23 +1,34 @@
-export declare type Options = { options: any; argv: string[] };
-export declare type Config = {
-	noAliasPropagation?: boolean | "first-only";
-	noCamelCase?: boolean;
-	noBundling?: boolean;
-	ignoreEquals?: boolean;
-	duplicates?:
-		| "use-first"
-		| "use-last"
-		| "limit-first"
-		| "limit-last"
-		| "error"
-		| "append"
-		| "stack"
-		| "stack-values";
-};
-export declare function getOpts(
+// Import the function like 
+// import getOpts  = require('get-options');
+// If you need the types you can import them like normal
+// import { Options } from 'get-options';
+
+/**
+ * Extract command-line options from a list of strings.
+ */
+declare function getOpts(
 	input: any[],
 	optdef?: string | { [key: string]: string },
-	config?: Config
-): Options;
+	config?: getOpts.Config
+): getOpts.Options;
 
-export default getOpts;
+export = getOpts;
+
+declare namespace getOpts {
+	export type Options = { options: any; argv: string[] };
+	export type Config = {
+		noAliasPropagation?: boolean |  "first-only";
+		noCamelCase?: boolean;
+		noBundling?: boolean;
+		ignoreEquals?: boolean;
+		duplicates?:
+			| "use-first"
+			| "use-last"
+			| "limit-first"
+			| "limit-last"
+			| "error"
+			| "append"
+			| "stack"
+			| "stack-values";
+	};
+}

